src/modules/keyboardmanager/test/KeyboardManagerRemapLogicTest.cpp

```diff
@@ -0,0 +1,1311 @@
+
+            // Send A keydown
+            mockedInputHandler.SendVirtualInput(nInputs, input, sizeof(INPUT));
+
+            // A key state should be true
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(0x41), true);
+            input[0].ki.dwFlags = KEYEVENTF_KEYUP;
+
+            // Send A keyup
+            mockedInputHandler.SendVirtualInput(nInputs, input, sizeof(INPUT));
+
+            // A key state should be false
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(0x41), false);
+        }
+    };
+
+    TEST_CLASS (SingleKeyRemappingTests)
+    {
+    public:
+        // Test if correct keyboard states are set for a single key remap
+        TEST_METHOD (RemappedKey_ShouldSetTargetKeyState_OnKeyEvent)
+        {
+            // Reset test environment
+            TestHelpers::ResetTestEnv(mockedInputHandler, testState);
+
+            // Set HandleSingleKeyRemapEvent as the hook procedure
+            std::function<intptr_t(LowlevelKeyboardEvent*)> currentHookProc = std::bind(&KeyboardEventHandlers::HandleSingleKeyRemapEvent, std::ref(mockedInputHandler), std::placeholders::_1, std::ref(testState));
+            mockedInputHandler.SetHookProc(currentHookProc);
+
+            // Remap A to B
+            testState.AddSingleKeyRemap(0x41, 0x42);
+            const int nInputs = 1;
+
+            INPUT input[nInputs] = {};
+            input[0].type = INPUT_KEYBOARD;
+            input[0].ki.wVk = 0x41;
+
+            // Send A keydown
+            mockedInputHandler.SendVirtualInput(1, input, sizeof(INPUT));
+
+            // A key state should be unchanged, and B key state should be true
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(0x41), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(0x42), true);
+            input[0].ki.dwFlags = KEYEVENTF_KEYUP;
+
+            // Send A keyup
+            mockedInputHandler.SendVirtualInput(1, input, sizeof(INPUT));
+
+            // A key state should be unchanged, and B key state should be false
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(0x41), false);
+            Assert::AreEqual(mockedInputHandler.GetVirtualKeyState(0x42), false);
+        }
+
+        // Test if key is suppressed if a key is disabled by single key remap
+        TEST_METHOD (RemappedKeyDisabled_ShouldNotChangeKeyState_OnKeyEvent)
+        {
+            // Reset test environment
+            TestHelpers::ResetTestEnv(mockedInputHandler, testState);
+
+            // Set HandleSingleKeyRemapEvent as the hook procedure
+            std::function<intptr_t(LowlevelKeyboardEvent*)> currentHookProc = std::bind(&KeyboardEventHandlers::HandleSingleKeyRemapEvent, std::ref(mockedInputHandler), std::placeholders::_1, std::ref(testState));
+            mockedInputHandler.SetHookProc(currentHookProc);
+
+            // Remap A to 0x0 (disabled)
+            testState.AddSingleKeyRemap(0x41, 0x0);
```

Comment:
What is meant by `disabled` here? Does it mean that the remapping is invalid or that the target key is empty?

Reply:
That's a nice and interesting feature. Thanks for the clarification on that!